tensorflow/compiler/jit/xla_compilation_cache.cc

```diff
@@ -187,21 +189,30 @@ Status XlaCompilationCache::BuildExecutable(
 
 Status XlaCompilationCache::Compile(
     const XlaCompiler::Options& options, const NameAttrList& function,
-    absl::Span<const XlaCompiler::Argument> args,
+    const std::vector<XlaCompiler::Argument>& args,
     const XlaCompiler::CompileOptions& compile_options,
     CompileMode compile_mode,
     const XlaCompiler::CompilationResult** out_compilation_result,
     xla::LocalExecutable** out_executable) {
-  absl::optional<int64> compile_threshold;
-  if (compile_mode == CompileMode::kLazy) {
-    compile_threshold = kDefaultCompilationThreshold;
-  }
-  auto compile_fn = [&](XlaCompiler* compiler,
+  // Explicitly capture all required data by value for async compilation
+  struct AsyncCompileData {
+    const XlaCompiler::CompileOptions& compile_options;
+    const NameAttrList& function;
+  } async_compile_data = {
+    compile_options, function
+  };
+
+  // Do not add additional variables to the capture list,
+  // but add any additional ones to the AsyncCompileData struct above
+  auto compile_fn = [async_compile_data](
+                        XlaCompiler* compiler,
+                        const std::vector<XlaCompiler::Argument>& args,
```

Comment:
I'm probably missing something, but wouldn't be simpler to add a

```
Status operator()(XlaCompiler* compiler, const std::vector<XlaCompiler::Argument>& args, XlaCompiler::CompilationResult* result);
```

to `AsyncCompileData` itself (and rename it to `AsyncCompileFunctor`)?

Reply:
how? XlaCompiler doesn't exist in this scope. neither does CompilationResult. Furthermore, args must be copied by value, which is why it's passed to CompileImpl.